tests/chainer_tests/test_reporter.py

```diff
@@ -163,27 +164,17 @@ def test_report_scope(self):
         self.assertNotIn('x', reporter.observation)
 
 
+@backend.inject_backend_tests(
+    ['test_basic', 'test_serialize_array_float', 'test_serialize_array_int'],
+    [{}, {'use_cuda': True}])
 class TestSummary(unittest.TestCase):
 
     def setUp(self):
         self.summary = chainer.reporter.Summary()
 
-    def test_numpy(self):
-        self.summary.add(numpy.array(1, 'f'))
-        self.summary.add(numpy.array(-2, 'f'))
-
-        mean = self.summary.compute_mean()
-        testing.assert_allclose(mean, numpy.array(-0.5, 'f'))
-
-        mean, std = self.summary.make_statistics()
-        testing.assert_allclose(mean, numpy.array(-0.5, 'f'))
-        testing.assert_allclose(std, numpy.array(1.5, 'f'))
-
-    @attr.gpu
-    def test_cupy(self):
-        xp = cuda.cupy
-        self.summary.add(xp.array(1, 'f'))
-        self.summary.add(xp.array(-2, 'f'))
+    def test_basic(self, backend_config):
+        self.summary.add(backend_config.get_array(numpy.array(1, 'f')))
```

Comment:
I think that `BackendConfig.get_array` is too general, at least for current backend candidates of Chainer. How about making `BackendConfig` have `xp` property that is either `numpy` or `cupy`?

Reply:
Okay.